### PR TITLE
Add declarative unmount

### PR DESCRIPTION
**Pull request by Mic92**

### CODE REVIEW

File: lib/types/mdadm.nix [hunk 1]
```diff
@@ -58,6 +58,17 @@
         lib.optionalAttrs (config.content != null) config.content._mount;
       # TODO we probably need to assemble the mdadm somehow
     };
+    _unmount = diskoLib.mkUnmountOption {
+      inherit config options;
+      default = lib.optionalAttrs (config.content != null) config.content._unmount;
+      # Why does this not work?
```
Comment (Mic92) [inline lib/types/mdadm.nix:64]: @Lassulus I don't quite understand why I cannot use `{ fs = ...; dev = ...; }` here.

**Review by pogobanane — APPROVED** [review]
from looking at it, looks good

**Review by Lassulus — APPROVED** [review]
changes lgtm, since this just adds new functionality and doesn't touch old one. I would be fine with testing this one "in production" :)

**Comment by Lassulus** [comment]
@mergify queue

**Comment by Mic92** [comment]
I actually want to make it a dependency in nixos-anywhere and disko-install

**Pull request merged**